hawkular-wildfly-agent/src/main/java/org/hawkular/agent/monitor/inventory/Operation.java

```diff
@@ -46,10 +48,15 @@
      * @param operationName the actual name of the operation as it is known to the actual resource being managed.
      *                      This is the name that is used when telling the managed resource what operation to invoke.
      *                      It may or may not be the same as <code>name</code>.
+     * @param params Additional params for this operation definition, e.g. coming from dmr describe-operation.
+     *                   Can be null.
      */
-    public Operation(ID id, Name name, L location, String operationName) {
+    public Operation(ID id, Name name, L location, String operationName, List<OperationParam> params) {
         super(id, name, location);
         this.operationName = operationName;
+        if (params!=null && !params.isEmpty()) {
+            addProperty("params", params);
```

Comment:
I would create a constant for "params" here in the Operation class - make it private for now if we don't need it elsewhere.